### PR TITLE
feat: move searchbar to navbar and make data explorer a view

### LINKED ISSUE #457: [TASK]: Move searchbar functionality from data explorer into Navbar; Make Data explorer its own view
**Describe the task**
Instead of having the data explorer be opened with the current button, make it its own view that is navigated to when user makes a search from the navbar.

**Task Items**
- [ ] Move data explorer into its own view
- [ ] Move functionality of data explorer search into navbar search
- [ ] Remove searchbar from data explorer view

### PR DESCRIPTION
**Pull request by echl**
# Description

Instead of having the data explorer be opened with the current button, make it its own view that is navigated to when user makes a search from the navbar.

https://user-images.githubusercontent.com/16358728/212147920-703b459f-76f7-4313-93c7-e1d14607e7e0.mov

Resolves #457

### CODE REVIEW

File: packages/client/hmi-client/src/router/index.ts [hunk 1]
```diff
@@ -21,6 +22,7 @@ export enum RoutePath {
 	Dataset = '/projects/:projectId/dataset/:assetId?',
 	Simulation = '/projects/:projectId/simulations/:assetId?',
 	SimulationResult = '/projects/:projectId/simulation-results/:assetId?',
+	DataExplorer = '/data-explorer',
```
Comment (YohannParis) [inline packages/client/hmi-client/src/router/index.ts:25]: maybe just `explorer`, to keep it simple

**Review by YohannParis — DISMISSED** [review]

File: packages/client/hmi-client/src/App.vue [hunk 2]
```diff
@@ -2,42 +2,20 @@
...
+import { RoutePath, useCurrentRoute } from './router/index';
 
 /**
  * Router
  */
 const route = useRoute();
-const { isCurrentRouteHome } = useCurrentRouter();
-const isSidebarVisible = computed(() => !isCurrentRouteHome.value);
+const currentRoute = useCurrentRoute();
+const isSidebarVisible = computed(() => currentRoute.value.path !== RoutePath.Home);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/App.vue:16]: Data Explorer should not display the sidebar as well.

**Comment by YohannParis** [comment]
We are missing the clear search features:
Screen Shot 2023-01-11 at 14 13 24

File: packages/client/hmi-client/src/components/Navbar.vue [hunk 3]
```diff
@@ -45,31 +45,29 @@ const userInitials = computed(() =>
...
-				class="data-explorer p-button p-button-icon-only p-button-rounded"
-				@click="emit('show-data-explorer')"
-				aria-label="Data Explorer"
-			>
-				<i class="pi pi-search" />
-			</Button>
+		<img src="@assets/images/logo.png" height="32" width="128" alt="logo" />
+		<section class="page-name">
+			<p>
+				{{ pageName }}
```
[the remark this answers is not in the document]
Comment (echl) [reply packages/client/hmi-client/src/components/Navbar.vue:59]: Will tackle this as part of implementing the navigation dropdown menu

[the remark this answers is not in the document]
Comment (echl) [reply packages/client/hmi-client/src/components/Navbar.vue:59]: Just a placeholder text for now.

**Comment by mwdchang** [comment]
If I refresh the page in the data-explorer, the search-bar's text is lost while the page reflects search query in the URL.

For example:
- At home, search for "covid"
- Hit enter to enter into the data explorer
- Refresh, the explorer still shows covid-related results, but the search bar is blank

**Comment by echl** [comment]
> We are missing the clear search features: Screen Shot 2023-01-11 at 14 13 24

Fixed!

**Comment by echl** [comment]
> If I refresh the page in the data-explorer, the search-bar's text is lost while the page reflects search query in the URL.
> 
> For example:
> 
> * At home, search for "covid"
> * Hit enter to enter into the data explorer
> * Refresh, the explorer still shows covid-related results, but the search bar is blank
> 
> LIkewise, if you search for "covid" then "abc", then page-back, the results will be covid-related but the search bar says "abc".

Fixed!

File: packages/client/hmi-client/src/components/data-explorer/search-bar.vue [hunk 4]
```diff
@@ -29,6 +34,7 @@ const route = useRoute();
 
 const searchText = ref('');
 const defaultText = computed(() => props.text);
+const isClearSearchButtonHidden = computed(() => !searchText.value);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-bar.vue:37]: Perfection! Love the attention to details

**Review by YohannParis — DISMISSED** [review]

**Review by YohannParis — APPROVED** [review]

**Pull request merged**